nndownload/nndownload.py

```diff
@@ -2005,170 +2005,134 @@ def download_video_comments(
     filename = replace_extension(filename, "comments.json")
 
     if os.path.exists(filename):
-        output(f"File {filename} exists. Skipping...\n", logging.INFO)
+        output(f"Comments file \"{filename}\" exists. Skipping...\n", logging.INFO)
         return False
 
-    last_time: int = _CMDL_OPTS.comments_from
-    # sys.maxsize is practically big enough (no videos on niconico have reached 100 mil comments)
-    comments_limit: int = sys.maxsize if _CMDL_OPTS.request_all_comments else _CMDL_OPTS.comments_limit
+    comments_from: int = _CMDL_OPTS.comments_from
+    comments_limit: int = _CMDL_OPTS.comments_limit
 
-    try:
+    if comments_from:
+        output(f"Requesting comments looking back from {comments_from} on each thread.\n", logging.INFO)
+    else:
+        # Default to the current time
+        comments_from = int(datetime.now(timezone.utc).timestamp())
+
+    if _CMDL_OPTS.all_comments:
+        output(f"Requesting all comments on each thread.\n", logging.INFO)
+        comments_limit = None
+    else:
+        output(f"Requesting up to {comments_limit} comments on each thread.\n", logging.INFO) # Defaults to 1000
+
+    with Progress(TextColumn("{task.description}"), BarColumn(), TaskProgressColumn(), TextColumn("{task.completed}/{task.total}"), transient=True) as progress:
+        tasks = []
         for thread in template_params["thread_params"]["targets"]:
-            thread_data = {
-                "commentCount": 0,
-                "comments": [],
-                "fork": thread["fork"],
-                "id": thread["id"]
-            }
-            COMMENTS_DATA_JSON["threads"].append(thread_data)
-
-            fetch_thread_comments(
-                session,
-                template_params["id"],
-                template_params["comment_server"],
-                template_params["thread_key"],
-                thread,
-                template_params["thread_params"]["language"],
-                last_time,
-                comments_limit,
-                thread_data
-            )
-        
-        output("Finished downloading comments for {0}.\n".format(template_params["id"]), logging.INFO)
+            comments_data = COMMENTS_DATA_JSON
+            thread = threading.Thread(target=fetch_comments_thread, args=(session, template_params["id"], template_params["comment_server"], template_params["thread_key"], thread, template_params["thread_params"]["language"], progress, comments_data, comments_from, comments_limit))
+            thread.start()
+            tasks.append({
+                "thread": thread,
+            })
+
+        for task in tasks:
```

Comment:
This follows our `download_hls` pattern. This way, we can fetch multiple threads at the same time and show each progress bar